src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -476,23 +478,25 @@ private boolean isViolatingEmptyLineBetweenFieldsPolicy(DetailAST detailAST) {
      * @return true, if token has empty two lines before and allowMultipleEmptyLines is false
      */
     private boolean hasNotAllowedTwoEmptyLinesBefore(DetailAST token) {
-        return !allowMultipleEmptyLines && hasEmptyLineBefore(token)
-                && isPrePreviousLineEmpty(token);
-    }
-
-    /**
-     * Checks if a token has empty pre-previous line.
-     * @param token DetailAST token.
-     * @return true, if token has empty lines before.
-     */
-    private boolean isPrePreviousLineEmpty(DetailAST token) {
         boolean result = false;
-        final int lineNo = token.getLineNo();
-        // 3 is the number of the pre-previous line because the numbering starts from zero.
-        final int number = 3;
-        if (lineNo >= number) {
-            final String prePreviousLine = getLines()[lineNo - number];
-            result = CommonUtils.isBlank(prePreviousLine);
+        if (!allowMultipleEmptyLines && hasEmptyLineBefore(token)) {
+            final List<EnumSet<LineFlag>> lineFlags = categorizeLines();
+            for (int i = token.getLineNo() - 2; i > 0 && !result; i--) {
+                final EnumSet<LineFlag> flags = lineFlags.get(i);
+                if (flags.contains(LineFlag.EMPTY)) {
+                    // 1 line above line being checked is empty.
+                    final EnumSet<LineFlag> previousLineFlags = lineFlags.get(i - 1);
+                    if (previousLineFlags.contains(LineFlag.EMPTY)) {
+                        // 2 lines above the line being checked
+                        // and after the previous token are empty.
+                        result = true;
```

Comment:
we should break out of the for loop as soon as we hit a result we are looking for.
there is nothing in this loop that will turn it back to `false`.

Reply:
@rnveach : I think I was doing this in a previous revision, but when I do the build fails with the Checkstyle violation:
```
SingleBreakOrContinue: Loops should not contain more than a single "break" or "continue" statement
```